tests/cpptests/test_cpp_iterator_index.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "gtest/gtest.h"
+#include "iterator_util.h"
+
+#include "src/forward_index.h"
+#include "src/iterators/inverted_index_iterator.h"
+
+typedef enum IndexType {
+    INDEX_TYPE_TERM_FULL,
+    INDEX_TYPE_NUMERIC_FULL,
+    INDEX_TYPE_TERM,
+    INDEX_TYPE_NUMERIC,
+    INDEX_TYPE_GENERIC,
+} IndexType;
+
+class IndexIteratorTest : public ::testing::TestWithParam<IndexType> {
+protected:
+    static constexpr size_t n_docs = 1234;
+    std::array<t_docId, n_docs> resultSet;
+    InvertedIndex *idx;
+    QueryIterator *it_base;
+
+    void SetUp() override {
+        // Generate a set of document IDs for testing
+        for (size_t i = 0; i < n_docs; ++i) {
+            resultSet[i] = 2 * i + 1; // Document IDs start from 1
+        }
+
+        switch (GetParam()) {
+            case INDEX_TYPE_TERM_FULL:
+                SetTermsInvIndex();
+                it_base = NewInvIndIterator_TermFull(idx);
+                break;
+            case INDEX_TYPE_NUMERIC_FULL:
+                SetNumericInvIndex();
+                it_base = NewInvIndIterator_NumericFull(idx);
+                break;
+            case INDEX_TYPE_TERM:
+                SetTermsInvIndex();
+                it_base = NewInvIndIterator_TermQuery(idx, nullptr, {true, RS_FIELDMASK_ALL}, nullptr, 1.0);
+                break;
+            case INDEX_TYPE_NUMERIC: {
+                SetNumericInvIndex();
+                FieldMaskOrIndex fieldMaskOrIndex = {.isFieldMask = false, .value = {.index = RS_INVALID_FIELD_INDEX}};
+                FieldFilterContext fieldCtx = {.field = fieldMaskOrIndex, .predicate = FIELD_EXPIRATION_DEFAULT};
+                it_base = NewInvIndIterator_NumericQuery(idx, nullptr, &fieldCtx, nullptr, -INFINITY, INFINITY);
+            }
+                break;
+            case INDEX_TYPE_GENERIC:
+                SetGenericInvIndex();
+                it_base = NewInvIndIterator_GenericQuery(idx, nullptr, 0, FIELD_EXPIRATION_DEFAULT);
+                break;
+        }
+    }
+    void TearDown() override {
+        it_base->Free(it_base);
+        InvertedIndex_Free(idx);
+    }
+
+private:
+    void SetTermsInvIndex() {
+        // This function should populate the InvertedIndex with terms
+        size_t memsize;
+        idx = NewInvertedIndex((IndexFlags)(INDEX_DEFAULT_FLAGS), 1, &memsize);
+        for (size_t i = 0; i < n_docs; ++i) {
+            ForwardIndexEntry h = {0};
+            h.docId = resultSet[i];
+            h.fieldMask = i + 1;
+            h.freq = i + 1;
+            h.term = "term";
```

Comment:
just validating, I understand here Iterating on an Inverted Index is validating on only one of the "entries" or "terms".

Reply:
We add dummy entries to the inverted index; they all have the same term (in theory, it could be another stemming like "terms", "Term", and so on), but for this test's sake, we don't care about the term, only about the ids and potentially the field mask